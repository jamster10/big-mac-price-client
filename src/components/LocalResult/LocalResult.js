import React from 'react'
import './LocalResult.css'



export default function LocalResult({currentCountry, amount}) {
  
  //calculate amount of big macs that could be bought locally
  const localBigMacCount = () => {
    const count = (amount / currentCountry.local_price)
    return `${count.toFixed(3)} Big Mac${count > 1 ? 's' : ''}`
  }

  return (
    <div>
      <p>You could buy {localBigMacCount()} in your country.</p>
      <p>Your Dollar Purchasing Parity (PPP) is {parseInt(currentCountry.dollar_ppp).toFixed(3)}.</p>  
    </div>
  )
}

LocalResult.defaultProps = {
  currentCountry: {
    dollar_ppp: null,
    local_price: null,
  },
  amount: null
}