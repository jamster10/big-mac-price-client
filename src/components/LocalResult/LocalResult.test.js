import React from 'react';
import ReactDOM from 'react-dom';
import LocalResult from './LocalResult';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<LocalResult />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<LocalResult/>)
    .toJSON();
  expect(tree).toMatchSnapshot();  
});
