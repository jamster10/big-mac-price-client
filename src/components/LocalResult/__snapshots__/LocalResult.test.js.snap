// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <p>
    You could buy 
    NaN Big Mac
     in your country.
  </p>
  <p>
    Your Dollar Purchasing Parity (PPP) is 
    NaN
    .
  </p>
</div>
`;
